src/utils/create-security-number-pad-array.ts

```diff
@@ -0,0 +1,57 @@
+interface PickRandomNumbersParams {
+  min?: number
+  max: number
+  pickCount: number
+  exceptions?: number[]
+}
+
+const pickRandomNumbers = ({
+  min = 0,
+  max,
+  pickCount,
+  exceptions = [],
+}: PickRandomNumbersParams) => {
+  const result: number[] = []
+  while (result.length < pickCount) {
+    const randomNumber = Math.floor(Math.random() * max) + min
+    if (!result.includes(randomNumber) && !exceptions.includes(randomNumber)) {
+      result.push(randomNumber)
+    }
+  }
+  return result
+}
+
+export interface CreateRandomNumberArrayParams {
+  numberPadCount?: number
+  totalPadCount?: number
+}
+
+export const EMPTY = 'EMPTY'
+export const DELETE = 'DELETE'
+
+export type PadValue = number | typeof EMPTY | typeof DELETE
+
+export const createSecurityNumberPadArray = ({
+  numberPadCount = 10,
+  totalPadCount = 15,
+}: CreateRandomNumberArrayParams): PadValue[] => {
+  const resultArray: PadValue[] = Array.from({
+    length: totalPadCount,
+  })
+  const randomEmptyIndex = pickRandomNumbers({
+    min: 0,
+    max: totalPadCount - 1,
+    pickCount: totalPadCount - numberPadCount - 1,
+  })
+  const randomNumberIndex = pickRandomNumbers({
+    min: 0,
+    max: totalPadCount - 1,
+    pickCount: numberPadCount,
+    exceptions: randomEmptyIndex,
+  })
+
+  randomEmptyIndex.forEach(val => (resultArray[val] = EMPTY))
+  randomNumberIndex.forEach((val, idx) => (resultArray[val] = idx))
```

Comment:
r:
- 숫자패드, EMPTY패드의 배열을 `shuffle` 함수를 구현해 섞는 방식은 어떨까요?
- `numberPadCount`는 항상 10일 것 같아 인자로 받는 의미가 있을까요? 그리드 빈칸 을 채울 EMPTY 패드 개수 계산을 위해 `totalPadCount`만 받아도 괜찮을 것 같습니다.
```suggestion
  /**
   * [0,1,...,8,9,EMPTY,EMPTY,EMPTY,EMPTY]
   */
  const pads = ...
  const shuffledPads = shuffle(pads);
```

Reply:
와! 덕분에 코드가 정말 간결해졌네요!

다시한번 설계의 중요성을 깨닫습니다